test/lit/wat-kitchen-sink.wast

```diff
@@ -1515,6 +1515,373 @@
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-func-indirect-index
+  block $l
+   try
+   delegate 0
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-func-indirect-name (type $void)
+ ;; CHECK-NEXT:  (block $l
+ ;; CHECK-NEXT:   (try
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (nop)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:    (delegate 1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-func-indirect-name
+  block $l
+   try
+   delegate $l
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-try-direct-index (type $void)
+ ;; CHECK-NEXT:  (block $label
+ ;; CHECK-NEXT:   (try $__delegate__label
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (try
+ ;; CHECK-NEXT:      (do
+ ;; CHECK-NEXT:       (nop)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (delegate $__delegate__label)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-try-direct-index
+  try
+   block
+    try
+    delegate 1
+   end
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-try-direct-name (type $void)
+ ;; CHECK-NEXT:  (block $l
+ ;; CHECK-NEXT:   (try $__delegate__l
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (try
+ ;; CHECK-NEXT:      (do
+ ;; CHECK-NEXT:       (nop)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (delegate $__delegate__l)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-try-direct-name
+  try $l
+   block
+    try
+    delegate $l
+   end
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-try-indirect-index (type $void)
+ ;; CHECK-NEXT:  (block $label
+ ;; CHECK-NEXT:   (try $__delegate__label
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (try
+ ;; CHECK-NEXT:      (do
+ ;; CHECK-NEXT:       (nop)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:      (delegate $__delegate__label)
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-try-indirect-index
+  try
+   block
+    try
+    delegate 0
+   end
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-try-indirect-name (type $void)
+ ;; CHECK-NEXT:  (block $label
+ ;; CHECK-NEXT:   (try $__delegate__label
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (block $l
+ ;; CHECK-NEXT:      (try
+ ;; CHECK-NEXT:       (do
+ ;; CHECK-NEXT:        (nop)
+ ;; CHECK-NEXT:       )
+ ;; CHECK-NEXT:       (delegate $__delegate__label)
+ ;; CHECK-NEXT:      )
+ ;; CHECK-NEXT:     )
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $try-delegate-nested-try-indirect-name
+  try
+   block $l
+    try
+    delegate $l
+   end
+  end
+ )
+
+ ;; CHECK:      (func $try-delegate-nested-try-shadowing (type $void)
+ ;; CHECK-NEXT:  (block $l
+ ;; CHECK-NEXT:   (try $__delegate__l
+ ;; CHECK-NEXT:    (do
+ ;; CHECK-NEXT:     (block $l_0
+ ;; CHECK-NEXT:      (block $l_1
```

Comment:
When do we create wrapping blocks? I thought we do when there are branches targeting the label. This case already has an inner block though... Why do we have two more blocks in the result?

Reply:
Because of how we handle delegate labels as prefixed versions of "normal" scope labels, the `delegate $l` here also sets the "normal" label for the `try $l` scope, so we end up emitting a wrapper block around the try as if there were a normal branch to `$l` as well.

We could add some logic to try to avoid this unnecessary wrapper block, but it didn't seem worth adding complexity to remove it.

In general the logic for adding wrapper blocks around a scope does not consider whether there are already blocks inside that scope.